mesonbuild/environment.py

```diff
@@ -395,6 +395,20 @@ def detect_cpu(compilers: CompilersDict) -> str:
     # detect_cpu_family() above.
     return trial
 
+kernel_mappings = {'freebsd': 'freebsd',
+                   'openbsd': 'openbsd',
+                   'windows': 'windows',
+                   'android': 'linux',
+                   'cygwin': 'windows',
+                   'darwin': 'xnu',
+                   }
```

Comment:
Hmm, I don't see linux in this mapping... does that mean that on Android, the .kernel() method returns "linux" but on desktop Linux it returns an error "Kernel not defined or could not be autodetected"?